src/Nut/Extensions.php

```diff
@@ -25,20 +25,20 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        if (count($this->app['extend.manager']->messages)) {
-            foreach ($this->app['extend.manager']->messages as $message) {
+        if (count($this->app['extend.manager']->getMessages())) {
+            foreach ($this->app['extend.manager']->getMessages() as $message) {
                 $output->writeln(sprintf('<error>%s</error>', $message));
             }
             return;
         }
 
-        $installed = $this->app['extend.manager']->showPackage('installed');
+        $allPackages = $this->app['extend.manager']->getAllPackages();
         $rows = [];
 
-        foreach ($installed as $ext) {
-            /** @var \Composer\Package\CompletePackageInterface $package */
-            $package = $ext['package'];
-            $rows[] = [$package->getPrettyName(), $package->getPrettyVersion(), $package->getType(), $package->getDescription()];
+        foreach ($allPackages as $type => $packages) {
+            foreach ($packages as $package) {
+                $rows[] = [$package['name'], $package['version'], $package['type'], $package['descrip']];
+            }
         }
 
         $table = $this->getHelper('table');
```

Comment:
With just the `->getMessages()` fix, my output is this:
```
+------------------+---------+----------------+-------------------------------------------------------+
| Name             | Version | Type           | Description                                           |
+------------------+---------+----------------+-------------------------------------------------------+
| bolt/boltbb      | v2.0.2  | bolt-extension | A bulletin board extension                            |
| bolt/clientlogin | v2.1.0  | bolt-extension | OAuth login for a Bolt site's clients in the frontend |
| bolt/geoip       | v2.0.2  | bolt-extension | Adds MaxMind GeoIP lookup ability to your templates   |
| bolt/members     | v2.1.1  | bolt-extension | Membership creation and management for Bolt           |
+------------------+---------+----------------+-------------------------------------------------------+
```

With the second part it error with this:
```
  [Symfony\Component\Routing\Exception\RouteNotFoundException]                           
  Unable to generate a URL for the named route "fileedit" as such route does not exist.  
```
And we very deliberately use objects.